staging/src/k8s.io/kubectl/pkg/cmd/create/create_namespace_test.go

```diff
@@ -17,45 +17,43 @@ limitations under the License.
 package create
 
 import (
-	"net/http"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"testing"
 
-	"k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/runtime/schema"
-	"k8s.io/cli-runtime/pkg/genericclioptions"
-	"k8s.io/client-go/rest/fake"
-	cmdtesting "k8s.io/kubectl/pkg/cmd/testing"
-	"k8s.io/kubectl/pkg/scheme"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
 )
 
 func TestCreateNamespace(t *testing.T) {
-	namespaceObject := &v1.Namespace{}
-	namespaceObject.Name = "my-namespace"
-	tf := cmdtesting.NewTestFactory()
-	defer tf.Cleanup()
-
-	codec := scheme.Codecs.LegacyCodec(scheme.Scheme.PrioritizedVersionsAllGroups()...)
-	ns := scheme.Codecs.WithoutConversion()
-
-	tf.Client = &fake.RESTClient{
-		GroupVersion:         schema.GroupVersion{Version: "v1"},
-		NegotiatedSerializer: ns,
-		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
-			switch p, m := req.URL.Path, req.Method; {
-			case p == "/namespaces" && m == "POST":
-				return &http.Response{StatusCode: http.StatusCreated, Header: cmdtesting.DefaultHeader(), Body: cmdtesting.ObjBody(codec, namespaceObject)}, nil
-			default:
-				t.Fatalf("unexpected request: %#v\n%#v", req.URL, req)
-				return nil, nil
-			}
-		}),
+	tests := map[string]struct {
```

Comment:
More tests can be found here, and of course you can add more test cases if needed.
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubectl/pkg/generate/versioned/namespace_test.go

Reply:
But not all the old test suite can be migrated into new one.Previous code uses a generator to generate a namespace and takes a map as paras which has a less stricted type check.

Reply:
yeap you are right :)

I don't think there's much more to test here, so I'll close this comment :)